astropy/units/format/latex.py

```diff
@@ -101,8 +101,8 @@ def format_exponential_notation(cls, val):
         -------
         latex_string : str
             The value in exponential notation in a format suitable for LaTeX.
-        """
-        if np.isfinite(val):
+        """#np.any() and np.all() which to use depends on the val array
+        if np.any(np.isfinite(val)):   #np.isfinite returns a booleans of array type so np.any(np.ndarray) returns true if any vaues in np.ndarray is true
```

Comment:
I don't think the comments are needed here - np.any etc. are already explained in the numpy documentation, we don't have to repeat that here.